src/firmware/boards/legacy_dongle_stm32f4/normal.c

```diff
@@ -447,27 +452,30 @@ static void send_drive_packet(const void *packet, const uint8_t *serials)
     // Message purpose
     mrf_write_long(address++, 0X0FU);
 
-    // Write out the payload sent from the host, interleaved with the prefix
-    // byte for each robot.
+    // Write out the payload sent from the host
     const uint8_t *rptr = packet;
-    for (size_t i = 0; i != NUM_ROBOTS; ++i)
+    for (size_t i = 0; i < packet_size; i++)
     {
-        mrf_write_long(address++,
-                       (serials[i] & 0x0F) | ((poll_index == i) ? 0x80 : 0x00));
-        for (size_t j = 0; j != DRIVE_BYTES_PER_ROBOT; ++j)
-        {
-            mrf_write_long(address++, *rptr++);
-        }
+        mrf_write_long(address++, *rptr++);
     }
-
-    // Write out the footer, which comprises the emergency stop status
+    // append the emergency stop status and feedback request robot id
     mrf_write_long(address++, estop_read() == ESTOP_RUN);
+    mrf_write_long(address++, poll_index);
 
     // Record the frame length, now that the frame is finished.
-    mrf_write_long(frame_length_address, address - header_start_address);
+    const size_t frame_length = address - header_start_address;
+    mrf_write_long(frame_length_address, frame_length);
+
+    // TODO: need to test this
+    // Do a final check that we're not going to exceed the max packet size
```

Comment:
TODO?

Reply:
Thanks for pointing this out. Leaving this here for now, will do a final validation pass before merging.

Reply:
bumping this